### PR TITLE
Consistent activity components (frequency, traits, Interact)

### PR DESCRIPTION
**Pull request by ebullient**

### CODE REVIEW

**Comment by MrVauxs** [comment]
Thank you for the fixes! However I must note, traits should go to the `traits` array, not bundled together with `components`

**Comment by ebullient** [comment]
the issue was how/when they were referenced... and how the sentence was being composed from that... (i.e. where traits were w/ Interact, or whatever, specifically.

**Comment by MrVauxs** [comment]
Its done automatically if you add them to the array of traits.

File: data/items/items-gmg.json [hunk 1]
```diff
@@ -824,7 +824,7 @@
 						"unit": "minute"
 					},
 					"components": [
-						"{@action Interact}"
+						"({@action Interact})"
```
Comment (ebullient) [inline data/items/items-gmg.json:827]: As a note, it's written/phrased this way here: https://2e.aonprd.com/Equipment.aspx?ID=612

**Comment by ebullient** [comment]
> Its done automatically if you add them to the array of traits.

Only if I'm using your *.js .. which I'm not. I'm digesting the JSON data to create something else (Obsidian markdown). So any behavior hidden in the code, rather than in the data, is problematic.

This construction is also a problem: 
```
"activity": {
    "number": 1,
    "unit": "varies",
    "entry": "{@as 3} command (emotion, enchantment,"
},
"components": [
    "incapacitation)"
],
```
As it implies this is read in a particular way to render a sentence, which may not be true. ;)

**Comment by MrVauxs** [comment]
> > Its done automatically if you add them to the array of traits.
> 
> Only if I'm using your *.js .. which I'm not. I'm digesting the JSON data to create something else (Obsidian markdown). So any behavior hidden in the code, rather than in the data, is problematic.
> 
> This construction is also a problem:
> 
> ```
> "activity": {
>     "number": 1,
>     "unit": "varies",
>     "entry": "{@as 3} command (emotion, enchantment,"
> },
> "components": [
>     "incapacitation)"
> ],
> ```
> 
> As it implies this is read in a particular way to render a sentence, which may not be true. ;)

This is just a parsing bug, and should be fixed accordingly.
We can't guarantee compatibility with 3rd-parties, and we are not making how traits and components are formatted in these into a string. You should worry about properly rendering the traits and components as opposed to irreversibly changing the data to fit the formatting.

**Comment by MrVauxs** [comment]
Overall, components should go into components, and traits into traits.
Formatting of these should be done automatically by the software rendering the data, because it is just data.

**Comment by ebullient** [comment]
If I understand what you're saying, you believe the syntax will always be: 
one or more components (comma-separated, no traits), followed by one or more traits (comma-separated, in parens).

In other words, only this is possible: envision, Interact (concentrate, manipulate, metamagic)
And never this: envision (concentrate), Interact (manipulate, metamagic)

Is that right?

**Comment by MrVauxs** [comment]
> If I understand what you're saying, the syntax will always be: one or more components (comma-separated, no traits), followed by one or more traits (comma-separated, in parens).
> 
> In other words, only this is possible: envision, Interact (concentrate, manipulate, metamagic) And never this: envision (concentrate), Interact (manipulate, metamagic)
> 
> Is that right? (If so, I can fix all of that now)

I meant that I don't care about formatting because that is handled by the aforementioned `.js` files.

What I do want is to have components and traits separated in separate arrays.

To follow your example, the JSON should look like:
```json
"components": [ "envision", "{@action Interact}" ],
"traits": [ "concentrate", "manipulate", "metamagic" ]
```

**Comment by ebullient** [comment]
> I meant that I don't care about formatting because that is handled by the aforementioned `.js` files.
> 
> To follow your example, the JSON should look like:
> 
> ```json
> "components": [ "envision", "{@action Interact}" ],
> "traits": [ "concentrate", "manipulate", "metamagic" ]
> ```

We're almost at total agreement. ;) I was asking for a clarification of what actually happens as I don't know Pf2e all that well.

Would traits ever be applied to components in this way: 
envision (concentrate), Interact (manipulate, metamagic)

I originally thought that would be possible, but I think you're telling me that it is not (which is why the separation between components and traits that you want works). I'm just hoping to confirm that what was possible in my head is actually not possible.

**Comment by ebullient** [comment]
I've made the updates (things moved to traits)

**Comment by MrVauxs** [comment]
> > I meant that I don't care about formatting because that is handled by the aforementioned `.js` files.
> > To follow your example, the JSON should look like:
> > ```json
> > "components": [ "envision", "{@action Interact}" ],
> > "traits": [ "concentrate", "manipulate", "metamagic" ]
> > ```
> 
> Can you just confirm that traits won't ever occur alongside components in this way: `envision (concentrate), Interact (manipulate, metamagic)`
> 
> I originally thought that would be possible, but I think you're telling me that it is not (which is why the separation between components and traits that you want works).

Yes, they never will, and neither is it a format ever used in the books.

**Pull request merged**